tex/chapters/Results.tex

```diff
@@ -2,12 +2,221 @@
 
 \chapter{Results}\label{cha:results}
 
-\section{Introduction}
+In this chapter, the implementation of a fingerprint function within the Axelrod-Python library will be examined.
+This includes the addition of two strategy transformers, the Dual and JossAnn as defined in \ref{}. %TODO
+Then several results will be presented where analytical fingerprints are compared with analytical ones.
+A discussion that compares different fingerprints of strategies within Axelrod-Python will also be given.
 
+\section{The Dual}
+The dual of a strategy is defined such that when when the original strategy and the dual are presented with identical histories they will return opposite actions.
+This relies on knowledge of how the original strategy would have behaved in a given situation, would be impractical to infer from the source code, however, the required behaviour can be achieved by having the original strategy as an attribute of the dual.
+Whenever the dual has to submit a move, it can first get the original strategy to suggest what move should it would have made, and then flip that action.
 
+\IncMargin{1.2em}
+\begin{algorithm}[H]
+ \KwData{A strategy}
+ \KwResult{The d of the strategy}{}
+  \If{First Turn}{
+   create copy of original strategy\;}
+  simulate original strategy\;
+  update original strategy's history/internal state\;
+  \Return{Flip of original strategy's move}
+ \caption{The Dual of a Strategy}
+\end{algorithm}\DecMargin{1.5em}
 
+\section{Implementation of Fingerprinting}\label{sec:fingerprint-implementation}
 
+As defined in section a fingerprint function is merely the expected score of a strategy when played against a Joss-Ann transformer of a probe with varying parameters. %TODO reference definition
+An approximation to this has been implemented within the Axelrod-Python library.
```

Comment:
As part of this project, a numerical implementation has now been included in the ...